docker-compose.build.yml

```diff
@@ -15,6 +15,7 @@ services:
     image: ethereumoptimism/builder:master
     volumes:
       - ./:/mnt
```

Comment:
Good thinking. Is the `$GOPATH` in the `docker-compose.build.yml` meant to reference the `GOPATH` in the `.env` file or on the host's machine? I recently encountered a bug where an env var in the `docker-compose` file was coming from the host machine when I intended it to be read from within the container

Reply:
If its meant to be the host machine, do you think there could be any problems that come with the hosts machine being a different architecture? I think we could use a docker volume so that running build once will download to the docker volume and then each time the docker volume is mounted in. This way the local gopath doesn't end up with differently compiled deps in the case of a different architecture

Reply:
Apologies for the delay on this @tynes , but circling back now. I had intended the `GOPATH` in `docker-compose.build.yml` to reference the one in the `docker-compose.env` file which is why I added it as a new env var  in there. I believe that should work as expected. I think we could possibly use a volume too such as mentioned in this SO post: https://stackoverflow.com/a/52484542. If you scroll up though you can see the accepted answer is more like the change I made. What do you think would be preferable?

Reply:
Here's another example if you scroll down a bit: https://evilmartians.com/chronicles/speeding-up-go-modules-for-docker-and-ci

In this example they're setting up a volume which sounds like what you were suggesting:

```
version: '3'
services:
  app:
    image: application:0.0.1-development
    build:
      context: .
      dockerfile: docker/development/Dockerfile
    volumes:
      - .:/app
      - go-modules:/go/pkg/mod # Put modules cache into a separate volume
  runner:
    <<: *app
  test:
    <<: *app
    command: go test ./...
volumes:
  go-modules: # Define the volume
```

Reply:
These changes have been made now @tynes